stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -171,7 +172,7 @@ public void onHalfClose() {
         method.invoke(request, responseObserver);
         request = null;
         responseObserver.freeze();
-        if (call.isReady()) {
+        if (!wasReady) {
```

Comment:
This is inverted. It should be `if (wasReady)`

Reply:
Oh, I thought you said onReady should only be called once, and it was which ever method (onHalfClose or onReady) was invoked first.  Is that not correct?

Reply:
Before `onHalfClose()`, `onReadyHandler == null`. So if there was an onReady callback, the code "ignored" it. The onReady() callback will generally happen before onHalfClose(). This check was to "replay" that missed onReady call.

Reply:
I see, updated.